returnn/torch/frontend/frontend.py

```diff
@@ -17,3 +24,78 @@ class TorchFrontend(Frontend[torch.Tensor]):
     """
 
     RawTensorType = torch.Tensor
+    _internal_frontend = TorchInternalFrontend
+
+    @staticmethod
+    def dot(a: _TT, b: _TT, *, reduce: Union[Dim, Sequence[Dim]]) -> _TT:
+        """
+        dot-product of a and b, see base class doc string
+        """
+        if isinstance(reduce, Dim):
+            reduce = [reduce]
+
+        a_dims = a.dims
+        b_dims = b.dims
+
+        assert all(
+            dim in a_dims for dim in reduce
+        ), f"'a' does not have the specified reduce dim(s) {reduce} (a dims: {a_dims})"
+        assert all(
+            dim in b_dims for dim in reduce
+        ), f"'b' does not have the specified reduce dim(s) {reduce} (b dims: {b_dims})"
+
+        reduce_dims_sorted_by_a = [dim for dim in a_dims if dim in reduce]
+
+        a_reduce_axes = [a_dims.index(reduce_dim) for reduce_dim in reduce_dims_sorted_by_a]
+        b_reduce_axes = [b_dims.index(reduce_dim) for reduce_dim in reduce_dims_sorted_by_a]
+
+        a_unique_axes = [i for (i, dim) in enumerate(a_dims) if dim not in b_dims]
```

Comment:
Naming ok? I found "unique" clearer than "var". But I can also make it consistent with what is used in `DotLayer`.